lib/resources/etc_hosts.rb

```diff
@@ -0,0 +1,82 @@
+# encoding: utf-8
+# author: Matthew Dromazos
+
+require 'utils/parser'
+require 'pry'
+class EtcHosts < Inspec.resource(1)
+  name 'etc_hosts'
+  desc 'Use the etc_hosts InSpec audit resource to find an
+    ip_address and its associated hosts'
+  example "
+  describe etc_hosts.where { ip_address == '127.0.0.1' } do
+    its ( 'ip_address' ) { should eq ['127.0.0.1'] }
+    its ( 'canonical_hostname' ) { should eq ['localhost'] }
+    its ( 'all_host_names' ) { should eq [['localhost', 'localhost.localdomain', 'localhost4', 'localhost4.localdomain4']] }
+  end
+  "
+
+  attr_reader :params
+
+  include CommentParser
+
+  def initialize(hosts_path = nil)
+    return skip_resource 'The `etc_hosts` resource is not supported on your OS.' unless inspec.os.linux? || inspec.os.windows?
+    @conf_path      = get_hosts_path_by_os(hosts_path)
+    @content        = nil
+    @params         = nil
+    read_content
+  end
+
+  filter = FilterTable.create
+  filter.add_accessor(:where)
+        .add_accessor(:entries)
+        .add(:ip_address,         field: 'ip_address')
+        .add(:canonical_hostname, field: 'canonical_hostname')
+        .add(:all_host_names,     field: 'all_host_names')
+
+  filter.connect(self, :params)
+
+  private
+
+  def get_hosts_path_by_os(hosts_path)
```

Comment:
I am unable to reproduce that, and likely had to do with the nil checking, not the structure of the conditional. We use this pattern in other resources already. You can see it at the top of the file resource.